### PR TITLE
changed: build opm-common python extension using cmake

### PR DESCRIPTION
**Pull request by akva2**
only use setup.py to generate wheels. adjust pypi packaging accordingly.

this unifies the handling with simulators, and make the whole process simpler and smoother. no syncing two different build systems, and we get rid of the bottleneck in the build process (setup.py always built using 1 build thread)..

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by bska** [comment]
Is this intended as an alternative to #3046 and/or #2369?

**Comment by bska** [comment]
Sorry&mdash;I accidentally hit the "close" button when I wanted to hit the "comment" button.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
orthogonal to #3043, however it makes #2369 obsolete.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
edited the PR description with my motivations.

**Comment by blattms** [comment]
I like the simplification, but it seems like some include paths might be missing. On a redhat system without system-installed pybind11 I get:
```
/opt/rh/devtoolset-9/root/usr/bin/g++  -DEMBEDDED_PYTHON -DFMT_HEADER_ONLY -DHAVE_CONFIG_H=1 -I/private/mabl/testbuildpython/opm/opm-common/build -I/private/mabl/testbuildpython/opm/opm-common/external/cjson -I/private/mabl/testbuildpython/opm/opm-common/build/include -isystem /private/mabl/testbuildpython/opm/opm-common/external/fmtlib/include -isystem /private/mabl/testbuildpython/opm/opm-common  -pipe -Wall -Wextra -Wshadow -fopenmp -pthread -O2 -DNDEBUG -UNDEBUG -fPIC   -fopenmp -std=c++17 -Wno-shadow -o CMakeFiles/opmcommon.dir/src/opm/input/eclipse/Schedule/Action/PyAction.cpp.o -c /private/mabl/testbuildpython/opm/opm-common/src/opm/input/eclipse/Schedule/Action/PyAction.cpp
In file included from /private/mabl/testbuildpython/opm/opm-common/src/opm/input/eclipse/Schedule/Action/PyAction.cpp:22:
/private/mabl/testbuildpython/opm/opm-common/src/opm/input/eclipse/Python/PyRunModule.hpp:26:10: fatal error: pybind11/embed.h: No such file or directory
   26 | #include <pybind11/embed.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -254,8 +254,7 @@ if (OPM_ENABLE_PYTHON)
   endif()
   find_package(pybind11 2.2 CONFIG)
   if (NOT pybind11_FOUND)
-    # Use full path for reuse with pypi
-    set(pybind11_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/python/pybind11/include ${PYTHON_INCLUDE_DIRS})
```
Comment (blattms) [inline CMakeLists.txt:258]: Might this be needed to support embedded python?

Comment (blattms) [reply CMakeLists.txt:258]: Seems like python/pybind11/CMakeLists.txt only sets the upper case version PYBIND11_INCLUDE_DIRS.

Comment (blattms) [reply CMakeLists.txt:258]: Indeed with this 
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4efb10..897655d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -255,6 +255,9 @@ if (OPM_ENABLE_PYTHON)
   find_package(pybind11 2.2 CONFIG)
   if (NOT pybind11_FOUND)
       add_subdirectory(python/pybind11)
+      if(NOT pybind11_INCLUDE_DIRS)
+        set(pybind11_INCLUDE_DIRS ${PYBIND11_INCLUDE_DIR} ${PYTHON_INCLUDE_DIRS})
+      endif()
   endif()
 endif()
 
```
it works and even fixes an issue with python on redhat for me.

**Review by blattms — CHANGES_REQUESTED** [review]
Looks good. Feel free to merge once my patch is in and jenkins approves.

File: CMakeLists.txt (hunk 1 above)
Comment (akva2) [reply CMakeLists.txt:258]: added.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
merging per prior approval.

**Pull request merged**